presto-main/src/main/java/com/facebook/presto/util/SpatialJoinUtils.java

```diff
@@ -168,4 +180,44 @@ private static boolean isSTDistance(RowExpression expression, FunctionManager fu
 
         return false;
     }
+
+    public static FunctionHandle getFlippedFunctionHandle(CallExpression callExpression, FunctionManager functionManager)
+    {
+        FunctionMetadata callExpressionMetadata = functionManager.getFunctionMetadata(callExpression.getFunctionHandle());
+        checkArgument(callExpressionMetadata.getOperatorType().isPresent());
+        OperatorType operatorType = flip(callExpressionMetadata.getOperatorType().get());
+        FunctionHandle functionHandle = functionManager.resolveOperator(
+                operatorType,
+                swapPair(fromTypes(callExpression.getArguments().stream().map(RowExpression::getType).collect(toImmutableList()))));
+        return functionHandle;
+    }
+
+    public static OperatorType flip(OperatorType operatorType)
```

Comment:
I have seen this helper in many other places. Can we move the common util to presto-expressions?

Reply:
Will do.

Reply:
I couldn't find a good class to put this in - any suggestions?